ckanext/scheming/validation.py

```diff
@@ -88,66 +90,118 @@ def validator(key, data, errors, context):
 
     return validator
 
+
+def validate_date_inputs(field, key, data, extras, errors, context):
+    date_error = _('Date format incorrect')
+    time_error = _('Time format incorrect')
+
+    date = None
+
+    def get_input(suffix):
+        inpt = key[0] + '_' + suffix
+        new_key = (inpt,) + tuple(x for x in key if x != key[0])
+        value = extras.get(inpt)
+        data[new_key] = value
+        errors[new_key] = []
+
+        if value:
+            del extras[inpt]
+
+        if field.get('required'):
+            not_empty(new_key, data, errors, context)
+
+        return (new_key, value)
+
+    date_key, value = get_input('date')
+    value_full = ''
+    if value != '':
```

Comment:
The problem I have is if I get rid of it, the validator doesn't get called at all it seems